src/main/java/net/dv8tion/jda/internal/handle/ChannelDeleteHandler.java

```diff
@@ -116,9 +122,16 @@ protected Long handleInternally(DataObject content)
 
                 guild.getStageChannelsView().remove(channel.getIdLong());
                 getJDA().handleEvent(
-                    new ChannelDeleteEvent(
-                        getJDA(), responseNumber,
-                        channel));
+                        new ChannelDeleteEvent(
+                                getJDA(), responseNumber,
+                                channel));
+
+                // Deleting any guild scheduled events associated to the deleted channel as they are deleted when the channel gets deleted.
+                // There is no delete event for the deletion of guild scheduled events, so we do this to keep the cache in sync.
+                guild.getScheduledEventsView().stream()
+                        .filter(guildScheduledEvent -> guildScheduledEvent.getType().equals(GuildScheduledEvent.Type.STAGE_INSTANCE) && guildScheduledEvent.getLocation().equals(String.valueOf(channelId)))
```

Comment:
You should use `Long.toUnsignedString` and do it *before* the stream, so it is not done for every event in the guild.